app/actions/v3/service_instance_delete.rb

```diff
@@ -194,8 +194,15 @@ def operation_in_progress!
         raise CloudController::Errors::ApiError.new_from_details('AsyncServiceInstanceOperationInProgress', service_instance.name)
       end
 
-      def unbinding_operation_in_progress!
-        raise UnbindingOperatationInProgress.new("An unbinding operation for a service binding of service instance #{service_instance.name} is in progress.")
+      def unbinding_operation_in_progress!(type, binding)
+        raise UnbindingOperatationInProgress.new(
+          case type
+          when :credential
+            "An operation for the service binding between app #{binding.app.name} and service instance #{service_instance.name} is in progress."
+          else
+            "An unbinding operation for a service binding of service instance #{service_instance.name} is in progress."
```

Comment:
I would make the two messages similar for consistency. So for `key bindings` I would say `An operation for the service binding of service instance  #{service_instance.name} is in progress.`

Reply:
Also, instead of passing `type` as an argument we can use `is_a` on the `binding` parameter as we have the whole object.